src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -716,8 +791,9 @@ private static List<String> getResponseBlobsToBeDeleted(ArmResponseBatchData arm
         if (nonNull(armResponseBatchData.getUploadFileFilenameProcessor())) {
             responseBlobsToBeDeleted.add(armResponseBatchData.getUploadFileFilenameProcessor().getUploadFileFilenameAndPath());
         }
-        if (nonNull(armResponseBatchData.getInvalidLineFileFilenameProcessor())) {
-            responseBlobsToBeDeleted.add(armResponseBatchData.getInvalidLineFileFilenameProcessor().getInvalidLineFileFilenameAndPath());
+        if (CollectionUtils.isNotEmpty(armResponseBatchData.getInvalidLineFileFilenameProcessors())) {
```

Comment:
Guessing this collection will never contain null entries?

Reply:
Its a list and is always created. Items are only added to the list.

Reply:
Is it possible to add null as an item to the list?

Reply:
Line 673 is the only place the IL responses are added and line 672 has a check for null before adding to the list so no, nulls wont get added.